Release note for NHC 4.10

abhatt-rh opened the pull request:
[TELCODOCS-344](https://issues.redhat.com/browse/TELCODOCS-344): Release note for Node Health Operator for OCP 4.10 (ECOPROJECT-250)

Preview: [Node Health Check Operator enhancements](https://deploy-preview-41743--osdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-10-release-notes.html#ocp-4-10-node-health-check-operator)

Note: The link in the release note will resolve correctly after the  https://github.com/openshift/openshift-docs/pull/41724 is merged.

abhatt-rh commented:
Hi @slintes and @prabinovRedhat ,
Request you to kindly review the release note for NHC 4.10. Thank you!

slintes commented on release_notes/ocp-4-10-release-notes.adoc:
```diff
@@ -290,6 +290,16 @@ For more information on configuration drift, see xref:../post_installation_confi
 You can now enable link:https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v2.html[Linux control groups version 2] (cgroups v2) on specific nodes in your cluster. The {product-title} process for enabling cgroups v2 disables all cgroups version 1 controllers and hierarchies. The {product-title} cgroups version 2 feature is in Developer Preview and is not supported by Red Hat at this time.
 
+[id="ocp-4-10-node-health-check-operator"]
+==== Node Health Check Operator enhancements
```
not sure if this would be the right place, but we need to mention somewhere that people, who are upgrading an existing installation of NHC, need to update the channel from `alpha` to `candidate` manually, otherwise the upgrade won't happen.

abhatt-rh replied:
@slintes I moved this out of release notes and added it to the [main docs PR](https://github.com/openshift/openshift-docs/pull/41724/files#diff-88f044abbb2615d4b62f74f1465525477e6571d9e193b910016a4b3892df2499R66-R73) to avoid confusion. Is that OK

mburke5678 commented on release_notes/ocp-4-10-release-notes.adoc:
```diff
@@ -334,6 +334,20 @@ For more information on configuration drift, see xref:../post_installation_confi
+
+The Node Health Check Operator provides these new enhancements:
+
+* Provides support for disconnected mode
```
s/Provides support/Support

Support for running in disconnected mode ?

mburke5678 commented on release_notes/ocp-4-10-release-notes.adoc:
```diff
@@ -334,6 +334,20 @@ For more information on configuration drift, see xref:../post_installation_confi
+The Node Health Check Operator provides these new enhancements:
+
+* Provides support for disconnected mode
+* Avoids conflicts with machine health check. For more information, see xref:../nodes/nodes/eco-node-health-check-operator.adoc#preventing-nhc-conflict-with-mhc_node-health-check-operator[Preventing node health check conflicts with machine health check]
```
If you change the heading in the other file, don't forget to update it here!

mburke5678 replied:
Another reminder: After you merge these PRs, you should double-check this link.

mburke5678 commented on release_notes/ocp-4-10-release-notes.adoc:
```diff
@@ -334,6 +334,20 @@ For more information on configuration drift, see xref:../post_installation_confi
+
+[NOTE]
+====
+To upgrade to the latest version of the Node Health Check Operator, you must manually change the channel name from `alpha` to `candidate`.
```
Will folks know where/how to set the channel name?

slintes replied:
Good question.
Maybe change to `... update the channel name in the Subscription...`? Should at least give a hint where that channel name is

abhatt-rh replied:
@mburke5678 and @slintes - I understand. I think based on your feedback, it might be appropriate to update it in the main docs where we mention [channel name in the Subscription YAML](https://github.com/openshift/openshift-docs/pull/41724/files#diff-88f044abbb2615d4b62f74f1465525477e6571d9e193b910016a4b3892df2499R66-R73). Hope that's ok.

mburke5678 replied:
Do we need a small module on how to update?

abhatt-rh replied:
@mburke5678 I am not sure if we would need a new module for changing a channel name since it is covered here. Maybe I could add these as additional resources. WDYT?

- [Changing the update channel for an Operator](https://docs.openshift.com/container-platform/4.9/updating/understanding-upgrade-channels-release.html#switching-between-channels_understanding-upgrade-channels-releases) 
- [Switching between channels](https://docs.openshift.com/container-platform/4.9/updating/understanding-upgrade-channels-release.html#switching-between-channels_understanding-upgrade-channels-releases)

mburke5678 commented:
@abhatt-rh A few small things. Otherwise LGTM

abhatt-rh commented on release_notes/ocp-4-10-release-notes.adoc:
```diff
@@ -290,6 +290,16 @@ For more information on configuration drift, see xref:../post_installation_confi
+
+* Avoids conflicts with machine health checks
+* Provides support for disconnected mode
+
```
@prabinovRedhat I think we can leave that out from release notes

abhatt-rh commented:
Hello @slintes @prabinovRedhat and @mburke5678 
Thank you all. I have addressed your feedback. Request you all to please take another look. If all looks OK, pls provide an _lgtm_

slintes commented:
lgtm

prabinovRedhat commented:
lgtm

prabinovRedhat commented:
lgtm

prabinovRedhat commented:
when I open this link https://github.com/openshift/openshift-docs/blob/main/nodes/nodes/eco-node-health-check-operator.adoc#how-nhc-prevent-conflict-with-mhc_node-health-check-operator I don't see related information (screenshot below)
![image](https://user-images.githubusercontent.com/58551443/155198218-53e958cf-92c4-487f-8170-f4554c04b3c7.png)

abhatt-rh commented:
> when I open this link https://github.com/openshift/openshift-docs/blob/main/nodes/nodes/eco-node-health-check-operator.adoc#how-nhc-prevent-conflict-with-mhc_node-health-check-operator I don't see related information (screenshot below) ![image](https://user-images.githubusercontent.com/58551443/155198218-53e958cf-92c4-487f-8170-f4554c04b3c7.png)
As mentioned in [my note in this comment](https://github.com/openshift/openshift-docs/pull/41743#issue-1131145247), the link will be active once the other PR is merged

prabinovRedhat commented:
lgtm
